### DISCUSSION TITLE
Fix proposal for Codesearch UI

### DISCUSSION THREAD

**Discussion by MaxRcd** [arnoudkooi/SN-Utils#511] [General]
Hello Arnoud,

First thing. Thank you a lot for this irreplaceable extension. Thanks to all contributors as well.

I am using Codesearch extensively. Sometimes I'm looking into records that don't have a name which leads the Codesearch UI to crash because a value checking is missing in the code.

**Error example:**
![Screenshot 2024-06-05 at 15 35 36](https://github.com/arnoudkooi/ServiceNow-Utils/assets/24252163/871da937-7ba3-456b-870f-cf078654926d)

**Example payload returned by the search API with null names:**
![Screenshot 2024-06-05 at 15 17 53](https://github.com/arnoudkooi/ServiceNow-Utils/assets/24252163/00946832-7fce-4b38-b65d-06d29bc0fe51)

I have fixed the code for my own usage. Fixed code is here : https://github.com/MaxRcd/ServiceNow-Utils/commit/598c493bbb08285cbb90e008ad377a4e19d426c6

I would be happy to have it in the actual browser extension so that I don't need to pull and fix the Codesearch UI everytime you release something new.

Regards

**Comment by arnoudkooi** [comment]
Thanks would be good to get fixed. 
Wouldn't `(hit?.name || '(empty)' ).replace(/<\/?[^>]+(>|$)/g, "")` fix it?

What does the second condition try to fix?

**Reply by MaxRcd** [reply]
Yes this is the short version. That seems fine to me since hit.name is only called once.

The second condition checks that hit.sysId and hit.matches are present. Like hit.name, their presence is assumed but not verified. If they are not present, there is no point in displaying the search result, hence the return statement. It's not a fix. it's more like a safety belt.

I will update the code with the short version.
Edit: here https://github.com/arnoudkooi/ServiceNow-Utils/compare/master...MaxRcd:ServiceNow-Utils:master?diff=unified&w=

**Comment by arnoudkooi** [comment]
Thanks, I'll add this in today's update. I don't have a strategy to check all property existence of existing API call payloads, so I'll leave that for now.

**Reply by Anduh** [reply]
This was fixed in [v.8.1.3.0 (2024-06-06)](https://github.com/arnoudkooi/SN-Utils/blob/master/CHANGELOG.md#8130-2024-06-06)

With v.10 released in April 2026, the [Code Search](https://snutils.com/docs/guide/slash-commands/search-commands#code-search) UI & backend was completely reworked, and I've had multiple searches where the results have been named as "unnamed", so would say this have stayed gone.
